semantic_router/encoders/vit.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, List, Optional
+
+from PIL import Image
+from PIL.Image import Image as _Image
+from pydantic.v1 import PrivateAttr
+
+from semantic_router.encoders import BaseEncoder
+
+
+class VitEncoder(BaseEncoder):
+    name: str = "google/vit-base-patch16-224"
+    type: str = "huggingface"
+    score_threshold: float = 0.5
+    processor_kwargs: dict = {}
+    model_kwargs: dict = {}
+    device: Optional[str] = None
+    _processor: Any = PrivateAttr()
+    _model: Any = PrivateAttr()
+    _torch: Any = PrivateAttr()
+    _T: Any = PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._processor, self._model = self._initialize_hf_model()
+
+    def _initialize_hf_model(self):
+        try:
+            from transformers import ViTImageProcessor, ViTModel
+        except ImportError:
+            raise ImportError(
+                "Please install transformers to use HuggingFaceEncoder. "
+                "You can install it with: "
+                "`pip install semantic-router[local]`"
+            )
+
+        try:
+            import torch
+            import torchvision.transforms as T
+        except ImportError:
+            raise ImportError(
+                "Please install Pytorch to use HuggingFaceEncoder. "
+                "You can install it with: "
+                "`pip install semantic-router[local]`"
+            )
+
+        self._torch = torch
+        self._T = T
+
+        processor = ViTImageProcessor.from_pretrained(
+            self.name, **self.processor_kwargs
+        )
+
+        model = ViTModel.from_pretrained(self.name, **self.model_kwargs)
+
+        if self.device:
+            model.to(self.device)
+
+        else:
+            device = "cuda" if self._torch.cuda.is_available() else "cpu"
+            model.to(device)
+            self.device = device
+
+        return processor, model
+
+    def _process_images(self, images: List[_Image]):
+        rgb_images = [self._ensure_rgb(img) for img in images]
+        processed_images = self._processor(images=rgb_images, return_tensors="pt")
+        processed_images = processed_images.to(self.device)
+        return processed_images
+
+    def _ensure_rgb(self, img: _Image):
+        rgbimg = Image.new("RGB", img.size)
+        rgbimg.paste(img)
+        return rgbimg
```

Comment:
Might want to have this as a utility when we integrate CLIP

Reply:
nice idea, after merging this to main I will merge it into the clip branch and move